File: src/main/java/demo/demo/model/Beneficiary.java
package demo.demo.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "BENEFICIARY")
@Data
public class Beneficiary {

    @Id
    @Column(name = "BENEFICIARY_ID")
    private Long beneficiaryId;

    @Column(name = "DNI_B", nullable = false, length = 8)
    private String dniB;

    @Column(name = "INSTITUTION_ID", nullable = false)
    private Long institutionId;

    @Column(name = "GENDER", nullable = false, length = 1)
    private String gender;

    @Column(name = "FIRST_NAME_B", nullable = false, length = 50)
    private String firstNameB;

    @Column(name = "LAST_NAME_B", nullable = false, length = 50)
    private String lastNameB;

    @Column(name = "BIRTH_DATE_B", nullable = false)
    private LocalDate birthDateB;

    @Column(name = "OCCUPATION", nullable = false, length = 20)
    private String occupation;

    @Column(name = "PHONE_B", nullable = false, length = 9)
    private String phoneB;

    @Column(name = "EMAIL_B", nullable = false, length = 35)
    private String emailB;
}

File: src/main/resources/static/js/beneficiary.js
document.addEventListener("DOMContentLoaded", function() {
    listarBeneficiarios();
});

function listarBeneficiarios() {
    fetch("http://localhost:8080/api/beneficiary/listar") // 🔹 nuevo endpoint
        .then(response => response.json())
        .then(data => {
            const tbody = document.getElementById("tablaBeneficiarios");
            tbody.innerHTML = "";
            if (!Array.isArray(data)) {
                console.error("La respuesta no es un array:", data);
                return;
            }
            data.forEach(b => {
                const row = document.createElement("tr");
                row.innerHTML = `
                    <td>${b.beneficiaryId}</td>
                    <td>${b.dniB}</td>
                    <td>${b.firstNameB}</td>
                    <td>${b.lastNameB}</td>
                    <td>${b.emailB}</td>
                    <td>${b.phoneB}</td>
                `;
                tbody.appendChild(row);
            });
        })
        .catch(err => console.error("Error listando beneficiarios:", err));
}

function guardarBeneficiario() {
    const b = {
        beneficiaryId: Number(document.getElementById("beneficiaryId").value),
        dniB: document.getElementById("dniB").value,
        institutionId: Number(document.getElementById("institutionId").value),
        gender: document.getElementById("gender").value,
        firstNameB: document.getElementById("firstNameB").value,
        lastNameB: document.getElementById("lastNameB").value,
        birthDateB: document.getElementById("birthDateB").value,
        occupation: document.getElementById("occupation").value,
        phoneB: document.getElementById("phoneB").value,
        emailB: document.getElementById("emailB").value
    };

    // Validar DNI antes de guardar
    fetch(`http://localhost:8080/api/beneficiary/dni/${b.dniB}`)
        .then(response => response.json())
        .then(existe => {
            if (existe) {
                alert("El DNI ya está registrado.");
            } else {
                fetch("http://localhost:8080/api/beneficiary/registrar", {
                    method: "POST",
                    headers: { "Content-Type": "application/json" },
                    body: JSON.stringify(b)
                })
                .then(res => res.text())
                .then(msg => {
                    alert(msg);
                    listarBeneficiarios(); // 🔹 refresca tabla
                })
                .catch(err => console.error("Error guardando beneficiario:", err));
            }
        });
}
